src/components/GraphiQL.js

```diff
@@ -650,34 +660,37 @@ export class GraphiQL extends React.Component {
 
   handleEditQuery = debounce(100, value => {
     if (this.state.schema) {
-      this._updateQueryFacts(value);
+      this.setState(
+        this._updateQueryFacts(
+          value,
+          this.state.operationName,
+          this.state.schema,
+        ),
+      );
     }
     this.setState({ query: value });
     if (this.props.onEditQuery) {
       return this.props.onEditQuery(value);
     }
   })
 
-  _updateQueryFacts = query => {
-    const queryFacts = getQueryFacts(this.state.schema, query);
+  _updateQueryFacts = (query, operationName, schema) => {
+    const queryFacts = getQueryFacts(schema, query);
     if (queryFacts) {
       // Update operation name should any query names change.
-      const operationName = getSelectedOperationName(
+      const updatedOperationName = getSelectedOperationName(
         this.state.operations,
-        this.state.operationName,
+        operationName,
         queryFacts.operations
       );
 
       // Report changing of operationName if it changed.
       const onEditOperationName = this.props.onEditOperationName;
-      if (onEditOperationName && operationName !== this.state.operationName) {
-        onEditOperationName(operationName);
+      if (onEditOperationName && operationName !== updatedOperationName) {
+        onEditOperationName(updatedOperationName);
       }
 
-      this.setState({
-        operationName,
-        ...queryFacts
```

Comment:
Yes exactly. Since it's spread in here, `updatedQueryAttributes.queryFacts` above won't exist, `updatedQueryAttributes` will contain the data spread in from the `queryFacts` var here